logstash-core/src/main/java/org/logstash/config/ir/graph/IfVertex.java

```diff
@@ -20,6 +21,9 @@ public BooleanExpression getBooleanExpression() {
 
     public IfVertex(SourceMetadata meta, BooleanExpression booleanExpression) {
         super(meta);
+        this.id = "if-" +
+                (meta != null ? Integer.toString(meta.hashCode()) : UUID.randomUUID().toString())
```

Comment:
This is intended to provide a consistent ID for if conditions. Maybe this isn't necessary, but it seems to be useful. Ideally we'd have a user-specified ID here.